### DISCUSSION TITLE
Proper way to cast types?

### DISCUSSION THREAD

**Discussion by dxu** [jrouwe/JoltPhysics.js#181] [Q&A] [answered]
Hi, what's the proper way to cast object types? 

For example, I'm trying to check the instance of a shape of a body, and then get the proper dimensions depending on the type. 

To do this, I'm using `getPointer` and `wrapPointer`, but it seems a bit circuitous, so I wanted to double check and make sure I'm not missing something obvious. 

This is what I'm currently doing:

```
const shape = body.GetShape();
// ...
  if (shape.GetSubType() === 3) {
    const capsuleShape = JOLT.wrapPointer(
      JOLT.getPointer(shape),
      JOLT.CapsuleShape,
    );
    const position = body.GetPosition();
    const rotation = body.GetRotation();
    const radius = capsuleShape.GetRadius();
    const halfHeight = capsuleShape.GetHalfHeightOfCylinder();
// ...
```

**Comment by LeXXik** [comment, marked as answer]
Yep, I always forget it, but its
```js
const actualObject = Jolt.castObject(somePointer, Jolt.ObjectConstructor);
```
